### ISSUE TITLE
Allow `"type": "module"` in `package.json`

### ISSUE THREAD

**Issue by SimonSimCity** [electron-vite/vite-plugin-electron#165]
I'm aware of that [this plugin does officially not support `"type": "module"`](https://github.com/electron-vite/vite-plugin-electron#be-aware), and I'm curious to how I could use it within a project, where (based on other plugins) have to set this option.

In my case, I've tried to run the project and renaming the files in the `electron` folder (`main.ts` and `preload.ts`) to the extension `cts`, which would enforce the compiler to make it `commonjs` compatible, but I couldn't get it to work ... The system I'm using prints out:
```ts
 ERROR  Unexpected token (Note that you need plugins to import files that are not JavaScript)                                                        12:36:00 PM
file: /Users/simon/Documents/my-nuxt-project/electron/main.cts:11:29
 9: let appReadyHasRun = false;
10: 
11: const navigateToUri = (window: BrowserWindow, url: string) => {
                                 ^
12:   window.webContents.send("route-changed", url);
13: };

 ERROR  Unexpected token (Note that you need plugins to import files that are not JavaScript)                                                        12:36:00 PM
file: /Users/simon/Documents/my-nuxt-project/electron/preload.cts:4:46
2: 
3: window.addEventListener("DOMContentLoaded", () => {
4:   ipcRenderer.on("route-changed", (_event, url: string) => {
                                                 ^
5:     window.dispatchEvent(new CustomEvent("route-changed", { detail: url }));
6:   });
```

How deep rooted is this limitation? Is this a limitation in Electron? Is it because of some drawbacks one would have?

**Comment by caoxiemeihao** [comment]
<img width="260" alt="image" src="https://github.com/electron-vite/vite-plugin-electron/assets/26263658/e837d8f2-b85d-4fd4-9f3b-4fb60c5a2aa9">

This limitation comes from `electron`. Will be support in [electron@28](https://github.com/electron/electron/issues/21457)

**Comment by SimonSimCity** [comment]
@caoxiemeihao thanks for this very welcome news, that `"type": "module"` is planned to be supported in electron by version 28. Since v27 is currently in beta, I expect this will still take a while until v28 is published.

In the meanwhile, is there a chance, that we could work around this issue, e.g. by defining our `main` and `preload` as `cts` files?

**Comment by caoxiemeihao** [comment]
Yep! `.mts` or `.mjs` always available in current version of electron.

**Comment by AlansCodeLog** [comment]
I'm using nuxt-electron which uses this plugin and I tried just changing to `.cts` but files were still compiled with a `.js` ending in `dist-electron` and if the package has `"type":"module"` it won't work because `.js` files will be assumed to be esm.

I found two workarounds, one is to override the output extension in the config.

```js
// for nuxt
electron: {
	build: [
		{
			entry: 'electron/main.ts',
			vite: {
			// for just vite
				build: {
					rollupOptions: {
						output: {
							entryFileNames: '[name].cjs',
							chunkFileNames: '[name].cjs',
						},
					},
				},
			},
		},
	],
},
````

Another simpler solution is to stick a `package.json` in `dist-electron` that says the folder is cjs, but you'd have to find a way to automate its creation.

```json
{
	"type": "commonjs"
}
```

**Comment by caoxiemeihao** [comment]
> I found two workarounds, one is to override the output extension in the config.

It's easier to use the `build.lib` option.

```js
electron({
  build: {
    lib: {
      entry: 'electron/main.ts',
      formats: ['cjs'],
      fileName: () => '[name].cjs',
    },
  },
})
```
